sdk/src/main/java/com/horizen/transaction/RegularTransactionCreator.java

```diff
@@ -14,26 +15,37 @@
 
 public class RegularTransactionCreator {
 
-    public static RegularTransaction create(NodeWallet wallet, List<Pair<PublicKey25519Proposition, Long>> to, PublicKey25519Proposition changeAddress, long fee, List<byte[]> boxIdsToExclude) {
+    public static RegularTransaction create(NodeWallet wallet, List<Pair<PublicKey25519Proposition, Long>> to,
+                                            List<Pair<MCPublicKeyHash, Long>> withdrawalRequests,
+                                            PublicKey25519Proposition changeAddress, long fee,
+                                            List<byte[]> boxIdsToExclude) {
         // 0. check parameters (fee >= 0, to.values >= 0, etc.)
         // 1. calculate sum of to.getValue(...) + fee
         // 2. get from wallet proper number of closed RegularBox ordered by creation time, which ids is not in boxIdsToExclude and sum of their values >= sum above
         // 3. set change to changeAddress if need
         // 4. construct inputs and outputs lists, timestamp
         // 5. try to do RegularTransaction.create(...)
 
-        if(wallet == null || to == null || changeAddress == null || boxIdsToExclude == null)
+        if(wallet == null || to == null || withdrawalRequests == null || changeAddress == null || boxIdsToExclude == null)
             throw new IllegalArgumentException("Parameters can't be null.");
-        long to_amount = 0;
+
+        long toAmount = 0;
         for(Pair<PublicKey25519Proposition, Long> pair : to) {
             if (pair.getValue() < 0)
                 throw new IllegalArgumentException("Output values must be >= 0.");
-            to_amount += pair.getValue();
+            toAmount += pair.getValue();
+        }
+
+        long withdrawalRequestsAmount = 0;
```

Comment:
why not to use the single aggregator `toAmount`? In general box regular and withdrawal outputs have the same meaning in context of coins